controllers/argocd/applicationset.go

```diff
@@ -943,7 +943,7 @@ func (r *ReconcileArgoCD) reconcileApplicationSetService(cr *argoproj.ArgoCD) er
 	}
 
 	svc.Spec.Selector = map[string]string{
-		common.ArgoCDKeyName: nameWithSuffix(common.ApplicationSetServiceNameSuffix, cr),
+		common.ArgoCDKeyName: argoutil.TruncateWithHash(fmt.Sprintf("%s-%s", cr.Name, common.ApplicationSetServiceNameSuffix)),
```

Comment:
```suggestion
		common.ArgoCDKeyName: nameWithSuffix(common.ApplicationSetServiceNameSuffix, cr),
```

This should be keeping the suffix as well.
For the long argocd name - `argocd-long-name-for-route-testiiiiiiiiiiiiiiiiiiiiiiiing` the selector with your current changes would have the value `argocd-long-name-for-route-testiiiiiiiiiiiiiiiii-2a814fa2e3b158` instead of the expected `argocd-long-name-for-r-2a814fa2e3b158-applicationset-controller` 
Correct me if wrong.

Reply:
No, I actually missed this!